Mathlib/Analysis/Convex/Gauge.lean

```diff
@@ -194,12 +194,22 @@ theorem self_subset_gauge_le_one : s ⊆ { x | gauge s x ≤ 1 } := fun _ => gau
 theorem Convex.gauge_le (hs : Convex ℝ s) (h₀ : (0 : E) ∈ s) (absorbs : Absorbent ℝ s) (a : ℝ) :
     Convex ℝ { x | gauge s x ≤ a } := by
   by_cases ha : 0 ≤ a
-  · rw [gauge_le_eq hs h₀ absorbs ha]
+  · rw [gauge_le_eq (hs.starConvex h₀) absorbs ha]
     exact convex_iInter fun i => convex_iInter fun _ => hs.smul _
   · -- Porting note: `convert` needed help
     convert convex_empty (𝕜 := ℝ) (E := E)
     exact eq_empty_iff_forall_not_mem.2 fun x hx => ha <| (gauge_nonneg _).trans hx
 
+theorem StarConvex.gauge_le (hs : StarConvex ℝ 0 s) (absorbs : Absorbent ℝ s) (a : ℝ) :
+    StarConvex ℝ 0 { x | gauge s x ≤ a } := by
+  by_cases ha : 0 ≤ a
+  · rw [gauge_le_eq hs absorbs ha]
+    refine starConvex_iInter fun i => starConvex_iInter fun _ => ?_
+    rw [← smul_zero i]
+    exact hs.smul i
```

Comment:
Maybe it would also be worth adding the lemma 
```
StarConvex.zero_smul {𝕜 E : Type *} [OrderedCommSemiring 𝕜] [AddCommMonoid E] [Module 𝕜 E] {x : E}
  {s : Set E} (hs : StarConvex 𝕜 0 s) (c : 𝕜) : StarConvex 𝕜 0 (c • s) := sorry
```
and then applying it here?